static/docs/commands-reference/pull.md

```diff
@@ -1,53 +1,123 @@
 hardlinks to put it into the working space without copying. See `dvc checkout`
 for more details.
 
-See `dvc remote`, `dvc config` and
-[remote storages](https://dvc.org/doc/get-started/configure)
-for more information on how to configure the remote storage.
+## Options
 
-```usage
-    usage: dvc pull [-h] [-q] [-v] [-j JOBS] [-r REMOTE] [-a]
-                    [-T] [-d] [-f]
-                    [targets [targets ...]]
+* `--show-checksums` Show checksums instead of file names.
 
-    positional arguments:
-      targets               DVC files.
+* `-r REMOTE`, `--remote REMOTE` specifies which remote cache 
+  (see `dvc remote list`) to pull from. The argument, `REMOTE`, is a
+  remote name defined using the `dvc remote add` command.
 
-    optional arguments:
-      -h, --help            show this help message and exit
-      -q, --quiet           Be quiet.
-      -v, --verbose         Be verbose.
-      -j JOBS, --jobs JOBS  Number of jobs to run simultaneously.
-      --show-checksums      Show checksums instead of file names.
-      -r REMOTE, --remote REMOTE
-                            Remote repository to pull from
-      -a, --all-branches    Fetch cache for all branches.
-      -T, --all-tags        Fetch cache for all tags.
-      -d, --with-deps       Fetch cache for all dependencies of the specified
-                            target.
-      -f, --force           Do not prompt when removing working directory files.
-      -R RECURSIVE, --recursive RECURSIVE
-                            Pull cache for subdirectories of the specified
-                            directory.
-```
+* `-a`, `--all-branches`  Determines the files to download by comparing across
+  all branches of the workspace.
+
+* `-T`, `--all-tags` Determines the files to download by comparing across
+  all tags of the workspace.
+
+* `-d`, `--with-deps` Determines the files to download by searching backwards
+  in the pipeline from the named stage(s).  The only files which will be
+  considered are associated with the named stage, and the stages which execute
+  earlier in the pipeline.
+
+* `-f`, `--force` Do not prompt when removing working directory files.
+
+* `-R dirname`, `--recursive dirname` Determines the files to download by
+  searching the named directory and its subdirectories for changed files.
+
+* `-j JOBS`, `--jobs JOBS` Number of jobs to run simultaneously while
+  downloading files from the remote cache.  The effect is to control the number
+  of files downloaded simultaneously.  For example with `-j 1` DVC downloads
+  one file at a time, with `-j 2` it downloads two at a time, and so forth.
+
+* `-h`, `--help` show the help message and exit
+
+* `-q`, `--quiet` do not write anything to standard output. Exit with 0 if
+  no problems arise, otherwise 1.
+
+* `-v`, `--verbose` displays detailed tracing information from executing the
+  `dvc pull` command.
 
 ## Examples
 
-Pull all files from the current Git branch:
+Define a named remote.
```

Comment:
So, I like the idea of showing how to define a remote, but in case of `dvc pull` usually remote is already defined in the repo you do `git clone` for. I would at least mention here in example. And may be in description?

Reply:
I thought that was necessary in order to set the readers baseline understanding of the context.  That section can be removed of course.

Reply:
I'm totally fine with setting the context. It's just the way it's done is a little bit misleading in this context. In most cases you don't actually have to define remote when you run `dvc pull`. It's already defined in the repository configuration and when you do `git clone` it pulls the definition for you. I don't mean that we need to remove this paragraph. I would just add some explanation.

Reply:
Right, there was quite a bit more that had to be said.